src/app/home/home-base/home-base.component.spec.ts

```diff
@@ -0,0 +1,96 @@
+import {HomeBaseComponent} from './home-base.component';
+import {ComponentFixture, TestBed} from '@angular/core/testing';
+import {YtVideoItem} from '../../_core/models';
+import {Observable, of} from 'rxjs';
+import {YoutubeDataService} from '../../_core/services/youtube-data.service';
+import {ServerStateService} from '../../_core/services/server-state.service';
+import {Component, Input} from '@angular/core';
+import {By} from '@angular/platform-browser';
+
+
+const YtVideoMock: YtVideoItem[] = [
+  {
+    videoId: 'dummy-id',
+    title: 'Angular in Darija is awesome',
+    description: 'Angular in Darija well explain in Darija',
+    thumbnailUrl: 'https://www.youtube.com/watch?v=ToaZsdRCc0s',
+    publishedAt: new Date().toString()
+  },
+  {
+    videoId: 'dummy-id-1',
+    title: 'Angular in Darija is awesome part 2',
+    description: 'Angular in Darija well explain in Darija',
+    thumbnailUrl: 'https://www.youtube.com/watch?v=ToaZsdRCc0s',
+    publishedAt: new Date().toString()
+  },
+];
+
+@Component({
+  selector: 'app-video-banner',
+  template: ''
+})
+export class VideoBannerStubComponent {
+  @Input() ytVideo: YtVideoItem | null = null;
+}
+
+@Component({
+  selector: 'app-video-listing',
+  template: ''
+})
+export class AppVideoListingStubComponent {
+  @Input() ytVideos: YtVideoItem[] = [];
+}
+
+@Component({
+  selector: 'app-ui-error',
+  template: 'Error'
+})
+export class ErrorStubComponent {
+  @Input() errorType!: string;
+}
+
+describe('HomeBaseComponent', () => {
+  let component: HomeBaseComponent;
+  let fixture: ComponentFixture<HomeBaseComponent>;
+  let youtubeDataService: any;
+  let serverStateServiceStub: Partial<ServerStateService>;
+  serverStateServiceStub = {
+    hydrate(key: string): (obs: Observable<any>) => Observable<any> {
+      return obs => {
+        return obs;
+      };
+    }
+  };
+  beforeEach(async () => {
```

Comment:
do we really need all TestBed stuff? wouldn't const home = new HomeBaseComponent for these test cases?

Reply:
I was testing the template with fixture.debugElement to see if the spinner was displayed, since I removed it, I removed the test and I am going to do as you advised just new HomeBaseComponent